lib/train/transports/ssh_connection.rb

```diff
@@ -316,8 +316,12 @@ def execute_on_channel(cmd)
         channel.exec(cmd) do |_, success|
           abort "Couldn't execute command on SSH." unless success
           channel.on_data do |_, data|
-            yield(data) if block_given?
-            stdout += data
+            if data == "Password: "
+              channel.send_data "#{@transport_options[:su_password]}\n"
```

Comment:
This should use the accessor.  Also it looks like any time the data contains "Password: " the su_password will be output, even if it is not set?  It also isn't clear to me why this isn't just done
in the caller by passing a block that looks for "Password: " and outputs the su_password, and why this needs to go into train.